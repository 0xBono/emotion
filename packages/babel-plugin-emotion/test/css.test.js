// @flow
import { createInlineTests } from './util'

const inline = {
  'label format with only local': {
    code: `
    import {css} from 'emotion'
    let cls = css({color:'hotpink'})
    `,
    opts: {
      labelFormat: 'my-css-[local]',
      autoLabel: true
    },
    filename: __filename
  },
  'label format with filename that is index and local': {
    code: `
    import {css} from 'emotion'
    let cls = css({color:'hotpink'})
    `,
    opts: {
      labelFormat: 'my-css-[filename]-[local]',
      autoLabel: true
    },
    filename: 'some-directory/index.js'
  },

  'label format with filename and local': {
    code: `
    import {css} from 'emotion'
    let cls = css({color:'hotpink'})
    `,
    opts: {
      labelFormat: 'my-css-[filename]-[local]',
      autoLabel: true
    },
    filename: __filename
  },

  'custom instance': {
    code: `
    import {css as lol} from 'my-emotion-instance'
    lol\`color:hotpink;\``,
    opts: {
      instances: ['my-emotion-instance']
    },
    filename: __filename
  },
  'custom instance relative': {
    code: `
    import {css as lol} from './my-emotion-instance'
    lol\`color:hotpink;\``,
    opts: {
      instances: ['./my-emotion-instance']
    },
    filename: __filename
  },
  'custom instance relative complex': {
    code: `
    import {css as lol} from '../test/my-emotion-instance'
    lol\`color:hotpink;\``,
    opts: {
      instances: ['./my-emotion-instance']
    },
    filename: __filename
  }
}
createInlineTests('babel css inline', inline)
