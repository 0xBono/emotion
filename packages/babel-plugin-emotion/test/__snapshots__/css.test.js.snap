// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`babel css inline babel 6 custom instance 1`] = `
"
import { css as lol } from 'my-emotion-instance';
/*#__PURE__*/lol('color:hotpink;');"
`;

exports[`babel css inline babel 6 custom instance relative 1`] = `
"
import { css as lol } from './my-emotion-instance';
/*#__PURE__*/lol('color:hotpink;');"
`;

exports[`babel css inline babel 6 custom instance relative complex 1`] = `
"
import { css as lol } from '../test/my-emotion-instance';
/*#__PURE__*/lol('color:hotpink;');"
`;

exports[`babel css inline babel 6 label format with filename and local 1`] = `
"import { css as _css } from 'emotion';

let cls = /*#__PURE__*/_css({
    name: '1wg0g17-my-css-css-test-cls',
    styles: 'color:hotpink;label:my-css-css-test-cls;'
});"
`;

exports[`babel css inline babel 6 label format with filename that is index and local 1`] = `
"import { css as _css } from 'emotion';

let cls = /*#__PURE__*/_css({
    name: '1uf5d32-my-css-some-directory-cls',
    styles: 'color:hotpink;label:my-css-some-directory-cls;'
});"
`;

exports[`babel css inline babel 6 label format with only local 1`] = `
"import { css as _css } from 'emotion';

let cls = /*#__PURE__*/_css({
    name: '1xiywpm-my-css-cls',
    styles: 'color:hotpink;label:my-css-cls;'
});"
`;

exports[`babel css inline babel 7 custom instance 1`] = `
"import { css as lol } from 'my-emotion-instance';

/*#__PURE__*/
lol(\\"color:hotpink;\\");"
`;

exports[`babel css inline babel 7 custom instance relative 1`] = `
"import { css as lol } from './my-emotion-instance';

/*#__PURE__*/
lol(\\"color:hotpink;\\");"
`;

exports[`babel css inline babel 7 custom instance relative complex 1`] = `
"import { css as lol } from '../test/my-emotion-instance';

/*#__PURE__*/
lol(\\"color:hotpink;\\");"
`;

exports[`babel css inline babel 7 label format with filename and local 1`] = `
"import { css as _css } from \\"emotion\\";

let cls =
/*#__PURE__*/
_css({
  name: \\"1wg0g17-my-css-css-test-cls\\",
  styles: \\"color:hotpink;label:my-css-css-test-cls;\\"
});"
`;

exports[`babel css inline babel 7 label format with filename that is index and local 1`] = `
"import { css as _css } from \\"emotion\\";

let cls =
/*#__PURE__*/
_css({
  name: \\"1uf5d32-my-css-some-directory-cls\\",
  styles: \\"color:hotpink;label:my-css-some-directory-cls;\\"
});"
`;

exports[`babel css inline babel 7 label format with only local 1`] = `
"import { css as _css } from \\"emotion\\";

let cls =
/*#__PURE__*/
_css({
  name: \\"1xiywpm-my-css-cls\\",
  styles: \\"color:hotpink;label:my-css-cls;\\"
});"
`;
